packages/grid/data-grid/src/tests/columnHeaders.DataGrid.spec.tsx

```diff
@@ -0,0 +1,53 @@
+import * as React from 'react';
+import { createClientRenderStrictMode } from 'test/utils';
+import { expect } from 'chai';
+import { DataGrid } from '@material-ui/data-grid';
+import { getColumnHeaderCell } from 'test/utils/helperFn';
+
+const isJSDOM = /jsdom/.test(window.navigator.userAgent);
+
+describe('<DataGrid /> - Column Headers', () => {
+  // TODO v5: replace with createClientRender
+  const render = createClientRenderStrictMode();
+
+  const baselineProps = {
+    autoHeight: isJSDOM,
+    rows: [
+      {
+        id: 0,
+        brand: 'Nike',
+      },
+      {
+        id: 1,
+        brand: 'Adidas',
+      },
+      {
+        id: 2,
+        brand: 'Puma',
+      },
+    ],
+  };
+
+  describe('headerClassName', () => {
+    it('should append the CSS class defined in headerClassName', () => {
+      render(
+        <div style={{ width: 300, height: 500 }}>
+          <DataGrid {...baselineProps} columns={[{ field: 'brand', headerClassName: 'foobar' }]} />
+        </div>,
+      );
+      expect(getColumnHeaderCell(0)).to.have.class('foobar');
+    });
+
+    it('should append the CSS class returned by headerClassName', () => {
+      render(
+        <div style={{ width: 300, height: 500 }}>
+          <DataGrid
+            {...baselineProps}
+            columns={[{ field: 'brand', headerClassName: () => 'foobar' }]}
+          />
+        </div>,
+      );
+      expect(getColumnHeaderCell(0)).to.have.class('foobar');
+    });
+  });
+});
```

Comment:
I'm confused. It's a TypeScript test but framed as a browser test. The browser test suite will never run it. I assumed you meant to write a browser test. How did you manage to assert that this test is running correctly?

Reply:
Ou, that is weird. I took it from a similar test for the `cellClassName`. I'll check what is going on.

Reply:
I wasn't clear. The extension of the file determines if it's a TypeScript test or a regular test case. spec.tsx vs. test.tsx

Reply:
Ok yes now I saw it, I added the wrong extension. I ran the tests and all passed and thought that it was ok since the test was fairly trivial.